packages/live-share-canvas/src/core/LiveCanvas.ts

```diff
@@ -425,8 +434,12 @@ export class LiveCanvas extends DataObject {
         liveStroke.clear();
     };
 
-    private getLocalUserInfo(): IUserInfo | undefined {
-        return this.onGetLocalUserInfo ? this.onGetLocalUserInfo() : undefined;
+    private getLocalEventUserInfo(): IEventUserInfo {
```

Comment:
Why change the name of this method? It's private anyway.

Reply:
This is where you should be calling `LiveShareClient.getClientInfo()` as the role of `getLocalUserInfo` is to build the `IUserInfo` object. You shouldn't do it in `updateCursorPosition`. If that means `getLocalUserInfo` needs to become async, that's fine.

Reply:
In James's changes, we are not broadcasting the `displayName` in the socket event anymore, since that is untrustworthy & easily spoofed. `LiveShareClient.getClientInfo()` checks all clientId (local or non-local), so `getLocalUserInfo` isn't really the same thing. He separated them because `LiveShareClient.getClientInfo()` does not yet support profile pictures, and thus that still gets broadcasted through the socket (though there is an argument for removing that until it does, since it is easily spoofable).

Reply:
also, the naming change was to make it clear how it was being used and what is being returned, which is now only a subset of `IUserInfo`

Reply:
Please see my comment below. I think you are just hiding a breaking change when you should explicitly introduce a breaking change that forces developers to update their code.

Reply:
resolved in meeting and other conversation thread